packages/shopping/src/application.ts

```diff
@@ -127,4 +145,43 @@ export class ShoppingApplication extends BootMixin(
 
     this.bind(UserServiceBindings.USER_SERVICE).toClass(MyUserService);
   }
+
+  async migrateSchema(options?: SchemaMigrationOptions) {
+    await super.migrateSchema(options);
+
+    const productRepo = await this.getRepository(ProductRepository);
+    await productRepo.deleteAll();
+    const productsDir = path.join(__dirname, '../fixtures/products');
+    const productFiles = fs.readdirSync(productsDir);
+
+    for (const file of productFiles) {
+      if (file.endsWith('.yml')) {
+        const productFile = path.join(productsDir, file);
+        const yamlString = fs.readFileSync(productFile, 'utf8');
+        const product = YAML.parse(yamlString);
+        await productRepo.create(product);
+      }
+    }
+
+    const passwordHasher = await this.get(
```

Comment:
you can inject the passwordhasher by `@inject(PasswordHasherBindings.PASSWORD_HASHER)`

Reply:
Will do in https://github.com/strongloop/loopback4-example-shopping/issues/478.

Reply:
I am not able to reproduce the browser popup. Let's revisit it during #478.